code/obj/item.dm

```diff
@@ -496,6 +507,13 @@
 						C.expose()
 				the_dir = turn(the_dir,45) */
 
+/obj/item/proc/combust_ended()
+	processing_items.Remove(src)
+	burning = null
+	firesource = 0
+	ClearSpecificOverlays("burn_overlay")
+	name = "[pick("charred","burned","scorched")] [name]"
```

Comment:
I think there's some kind of an obj prefix system we have that can handle this better. Does anyone know how that thing works?

Reply:
look at code for `piss-stained` or `bloody` and it should come up

Reply:
okies :heart:

Reply:
this is tied to clothing specifically and also runs through a can_stain flag, so moving that over to the global item properties would be a huge chore...Would you mind if I leave this line as is?